### PR TITLE
Add multi providers email support

### PR DESCRIPTION
**Pull request by mygalus**
Add multi-provider email support #3782
Google Part

### CODE REVIEW

File: packages/server/email/provider/google/googleDriver.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+const mailGoogle = (getGlobalConfig().provider == 'google')? new nodemailer.createTransport({
+  service: 'gmail',
+  auth: {
+         user: config.user,
+         pass: config.pass
+     }
+ }): undefined
+
+if (mailGoogle) {
+  mailGoogle.sendEmail = function (mailOptions) {
```
Comment (mattkrick) [inline packages/server/email/provider/google/googleDriver.js:16]: I like what you're doing here!
In this case, I might refactor this into an  `abstract class MailProvider` that is implemented by a `MailProviderGoogle` and `MailProviderMailgun`. That way the `sendMail` contract is clear and if/when we need to add another provider it'll be clear what's needed

**Comment by mattkrick** [comment]
Thank you so much for this!
I _really_ like it! Our email patterns haven't been touched in a very long time & this is the perfect excuse to give them some more love to support other providers. Don't worry about the merge conflicts, I can take care of those

File: packages/server/email/provider/google/googleDriver.js (hunk 1 above)
Comment (guillaume-tmx) [reply packages/server/email/provider/google/googleDriver.js:16]: I will look at it, Thanks

Comment (mygalus) [reply packages/server/email/provider/google/googleDriver.js:16]: Thanks for your feedbak. I will try to do it.

**Comment by mattkrick** [comment]
all patched up and merged to master in #3838! Thanks @mygalus these fixes are scheduled to go live tomorrow

**Pull request closed without merging**